### PR TITLE
Add docker-compose scripts for Docker development

### PR DESCRIPTION
**Pull request by francbartoli**

### CODE REVIEW

**Comment by francbartoli** [comment]
@afabiani can we merge this? Then I can port it to geonode base code

**Comment by afabiani** [comment]
Yes ok, I did not test it yet though.

**Comment by frafra** [comment]
Ops sorry, I have been busy. What about sharing the variables between different sections?

File: docker-compose.development.override.yml [hunk 1]
```diff
@@ -0,0 +1,37 @@
...
+      ALLOWED_HOSTS: "['localhost']"
+      GEOSERVER_PUBLIC_LOCATION: http://localhost/gs/
+
+  celery:
+    command: celery worker --app={{project_name}}.celeryapp:app --broker=amqp://guest:guest@rabbitmq:5672/ -B -l INFO
+    volumes:
+      - .:/usr/src/{{project_name}}
+      - statics:/mnt/volumes/statics
+      - geoserver-data-dir:/geoserver_data/data
+    environment:
```
Comment (frafra) [inline docker-compose.development.override.yml:24]: What about https://docs.docker.com/compose/compose-file/#extension-fields?

**Comment by francbartoli** [comment]
ok then we can wait for your tests. I believe @Coop56 has some feedbacks from his tests

**Comment by francbartoli** [comment]
@frafra 
> Ops sorry, I have been busy. What about sharing the variables between different sections?

do you mean fill .env root file with shared variables and inject them into different compose files? Or something different?

**Comment by frafra** [comment]
@francbartoli no, there's a different way to do it, that is currently used by spcgeonode. I put a link in the code review.

**Comment by francbartoli** [comment]
@frafra do you mean https://github.com/GeoNode/geonode/blob/master/scripts/spcgeonode/docker-compose.override.yml#L5?

**Comment by frafra** [comment]
@francbartoli Yes, sorry, I didn't submitted the review :)

File: docker-compose.development.override.yml (hunk 1 above)
Comment (francbartoli) [reply docker-compose.development.override.yml:24]: They look like interesting but also they need an upgrade of the docker-compose format version to at least 2.4. I'd rather to put this enhancement into a separated PR if you agree

Comment (frafra) [reply docker-compose.development.override.yml:24]: I agree with you.

**Pull request merged**